test/unit/test_row_factory.py

```diff
@@ -0,0 +1,54 @@
+from unittest import mock
+import pytest
+from etlhelper.row_factories import namedtuple_rowfactory
+import sqlite3
+
+
+# Test uses sqlite3 cursor, but all dbapi compliant cursors will be the same.
+mock_cursor = mock.MagicMock(sqlite3.Cursor)
+mock_cursor.fetchmany.return_value = [
+    (1, 2, 3),
+    (4, 5, 6),
+]
+
+
+def test_valid_field_names():
+    mock_cursor.description = (('id', None, None, None, None, None, None),
+                               ('name', None, None, None, None, None, None),
+                               ('desc', None, None, None, None, None, None))
+
+    create_row = namedtuple_rowfactory(mock_cursor)
+
+    rows = (create_row(row) for row in mock_cursor.fetchmany(50))
+
+    for i, row in enumerate(rows):
+        assert row._fields[0] == "id"
+        assert row._fields[1] == "name"
+        assert row._fields[2] == "desc"
+
+        assert row[0] == mock_cursor.fetchmany.return_value[i][0]
```

Comment:
The cool thing about namedtuples is they can be equal to normal tuples, so you can test all this with:

`assert rows == FAKE_DATA`